src/framework/theme/components/chat/chat-custom-message.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { Directive, Input, OnDestroy, OnInit, TemplateRef } from '@angular/core';
+import { NbCustomMessageService } from './custom-message.service';
+
+function throwCustomMessageTypeIsRequired(): void {
+  throw new Error('[nbCustomMessage]: custom message type is required.');
+}
+
+/**
+ * Usage details:
+ * [nbCustomMessage]: should be applied to the ng-template
+ * or use a structural directive syntax:
+ *
+ * <some-element *nbCustomMessage="customTypeName">
+ * <yourCustomTemplate></yourCustomTemplate>
+ * </some-element>
+ */
+
+@Directive({
+  selector: `[nbCustomMessage]`,
+})
+export class NbChatCustomMessageDirective implements OnInit, OnDestroy {
```

Comment:
Re-export in the `theme/public_api.ts` file so users will be able to import the class from the `@nebular/theme`